File: baekjoon_all/01000+/boj_1021.py
# Solve 2023-03-01
# Update 2023-10-14

import sys
from collections import deque

input = lambda : sys.stdin.readline().rstrip()

n, m = map(int, input().split())
targets = map(int, input().split())

dque = deque(range(1, n + 1))
ans = 0

for target in targets:
    cnt = 0

    while dque[0] != target:
        dque.append(dque.popleft())
        cnt += 1

    ans += min(cnt, len(dque) - cnt)

    dque.popleft()

print(ans)

File: baekjoon_all/01000+/boj_1021.cpp
// Solve 2022-06-06
// Update 2023-10-14

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

int main() {
    FASTIO;

    int n, m;
    cin >> n >> m;

    deque<int> dque;

    for (int i = 1; i <= n; i++) {
        dque.push_back(i);
    }

    int ans = 0;

    for (int i = 0; i < m; i++) {
        int x;
        cin >> x;

        int cnt = 0;

        while (dque.front() != x) {
            dque.push_back(dque.front());
            dque.pop_front();
            cnt++;
        }

        ans += min(cnt, SIZE(dque) - cnt);

        dque.pop_front();
    }

    cout << ans << '\n';

    return 0;
}
